### DISCUSSION TITLE
V3 ideas

### DISCUSSION THREAD

**Discussion by xhit** [xhit/go-simple-mail#53] [Ideas]
Discussion to get all your ideas to make this package better in V3

**Comment by jaskerv** [comment]
This ticket! https://github.com/xhit/go-simple-mail/issues/44

**Reply by xhit** [reply]
> This ticket! https://github.com/xhit/go-simple-mail/issues/44

Maybe some helpful methods:

GetBody
GetFrom,CC,BCC, etc..
GetAttachments (maybe a struct with IsInline variable)

Any other? Feel free to share :)

**Comment by IceflowRE** [comment]
#45 Should be fixed.

**Comment by liampulles** [comment]
#26 Would be great for our team - using context.Context is great for deadlines and cancellations with a system that has a lot of service calls - I can cancel a context in the root service and it will permeate right down to final service in the chain. I think it has also become the "idiomatic" way to handle timeouts and the like. :)

**Comment by Foosec** [comment]
Adding PGP support would be a nice-to.have. More widespread usage of services having the option to give them your public key, can only be a good thing. Google can scan the ciphertext all they want :)

**Comment by ganigeorgiev** [comment]
Currently it is not possible to pass a mock `SMTPClient` implementation to `Email.Send` and practically there is no easy way to run your app unit tests without connecting to actual email server or creating complex app level wrappers and therefore I want to suggest reviving #32 by creating interfaces so that the library components could be mocked more easily.

**Comment by xhit** [comment]
Hello to all, any feature, feel free to open a PR :) thanks for using this package https://github.com/xhit/go-simple-mail/issues/60
